### PR TITLE
Refactor individual Chats to Group Chats and UI improvements

### PR DESCRIPTION
**Pull request by MarcelTorne**

### CODE REVIEW

**Review by tberkane — COMMENTED** [review]
Great job once again! Only small nitpicks to make

File: app/src/main/java/com/github/epfl/meili/messages/ChatLogActivity.kt [hunk 1]
```diff
@@ -8,88 +9,129 @@ import android.widget.TextView
...
-import com.google.firebase.ktx.Firebase
 import com.xwray.groupie.GroupAdapter
 import com.xwray.groupie.GroupieViewHolder
 import com.xwray.groupie.Item
 
 class ChatLogActivity : AppCompatActivity() {
 
-
     companion object {
         val TAG = "ChatLog"
```
[the remark this answers is not in the document]
Comment (aman3014) [reply app/src/main/java/com/github/epfl/meili/messages/ChatLogActivity.kt:26]: Actually, you can't use this along with `const val`.

File: app/src/main/java/com/github/epfl/meili/messages/ChatLogActivity.kt [hunk 2]
```diff
@@ -8,88 +9,129 @@ import android.widget.TextView
...
 
-        val poi = intent.getParcelableExtra<PointOfInterest>("POI_KEY")
-        supportActionBar?.title = poi?.name
-
-
-        val groupId: String = poi?.placeId!!
+        verifyAndUpdateUserIsLoggedIn(Auth.isLoggedIn.value!!)
+    }
 
+    fun verifyAndUpdateUserIsLoggedIn(isLoggedIn: Boolean) {
```
[the remark this answers is not in the document]
Comment (MarcelTorne) [reply app/src/main/java/com/github/epfl/meili/messages/ChatLogActivity.kt:49]: For more information to not repeat this mistakes, what is the convention behind changing isLoggedIn to loggedIn?

File: app/src/androidTest/java/com/github/epfl/meili/models/ChatMessageUnitTest.kt [hunk 3]
```diff
@@ -1,26 +1,24 @@
...
-import com.google.firebase.firestore.DocumentSnapshot
 import org.hamcrest.CoreMatchers.`is`
+import org.junit.Assert.assertThat
 import org.junit.Test
 
-import org.junit.Assert.*
-
 
 class ChatMessageUnitTest {
     private val TEST_ID: String = "1"
```
[the remark this answers is not in the document]
Comment (MarcelTorne) [reply app/src/androidTest/java/com/github/epfl/meili/models/ChatMessageUnitTest.kt:10]: I get the error `Const 'val' are only allowed on top level or in objects` if I add const

File: app/src/androidTest/java/com/github/epfl/meili/messages/FirebaseMessageDatabaseAdapterTest.kt [hunk 4]
```diff
@@ -34,15 +35,40 @@ class FirebaseMessageDatabaseAdapterTest {
...
         val observer = Observer { _: Observable, _: Any ->
-            assertEquals(expectedList, db.messages)
+            assertEquals(expectedList, db.getMessages())
         }
 
         db.addObserver(observer)
 
         db.addMessageToDatabase(fake_message)
     }
+
```
[the remark this answers is not in the document]
Comment (MarcelTorne) [reply app/src/androidTest/java/com/github/epfl/meili/messages/FirebaseMessageDatabaseAdapterTest.kt:49]: I didn't find a way to create a DataSnapshot, I will add a comment regarding this. Please let me know if you do and I'll add this.

File: app/src/androidTest/java/com/github/epfl/meili/home/MockAuthenticationService.kt [hunk 5]
```diff
@@ -2,20 +2,18 @@ package com.github.epfl.meili.home
 
 import android.app.Activity
 import android.content.Intent
+import com.github.epfl.meili.models.User
 
 class MockAuthenticationService : AuthenticationService {
     var mock_name = "Meili User"
```
[the remark this answers is not in the document]
Comment (MarcelTorne) [reply app/src/androidTest/java/com/github/epfl/meili/home/MockAuthenticationService.kt:9]: I added a comment why they are not constants (I am modifying its value in the tests)

**Review by Bwananah — APPROVED** [review]

File: app/src/main/java/com/github/epfl/meili/models/User.kt [hunk 6]
```diff
@@ -9,6 +9,7 @@ import kotlinx.parcelize.Parcelize
 data class User(
     val uid: String,
     val username: String,
+    val email: String
 ) : Parcelable {
```
Comment (dyxcooh) [inline app/src/main/java/com/github/epfl/meili/models/User.kt:13]: Just a question about the choice here, why did you choose to use Parcelable to implement User?

**Review by dyxcooh — APPROVED** [review]
Great job! Your code is clean and self-explanatory

File: app/src/main/java/com/github/epfl/meili/models/User.kt (hunk 6 above)
Comment (MarcelTorne) [reply app/src/main/java/com/github/epfl/meili/models/User.kt:13]: Using Parcelable allows us to save it and retrieve it from FirebaseDatabase more easily

**Pull request merged**